ngraph/core/reference/src/runtime/opt_kernel/timer.cpp

```diff
@@ -0,0 +1,128 @@
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "ngraph/runtime/opt_kernel/timer.hpp"
+#include <algorithm>
+#include <iomanip>
+#include <iostream>
+#include <sstream>
+#include <vector>
+
+namespace ngraph
+{
+    namespace perf
+    {
+        Timer::Timer(std::string timer_name)
+            : m_name{std::move(timer_name)}
+        {
+            m_time_blocks.emplace_back("total");
+            m_time_blocks.front().start();
+        }
+
+        void Timer::finish()
+        {
+            // end the total time measurement for this timer
+            m_time_blocks.front().end();
+
+            std::stringstream total_duration;
+            total_duration << std::setw(10) << std::fixed << m_time_blocks.front().elapsed_time()
+                           << " [s]";
+
+            std::vector<std::string> output;
+            if (m_time_blocks.size() == 1)
+            {
+                const std::string timer_name =
+                    "  Total execution time for timer \"" + m_name + "\": " + total_duration.str();
+                const std::string separator_line(timer_name.length() + 2, '=');
```

Comment:
```suggestion
                const std::string separator_line(timer_name.length(), '=');
```

Reply:
This is intentional padding with 2 spaces at the end of the line. This util will be removed though.